urlconnection/src/main/java/zipkin/reporter/urlconnection/URLConnectionReporter.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright 2016 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.reporter.urlconnection;
+
+import com.google.auto.value.AutoValue;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.zip.GZIPOutputStream;
+import zipkin.Codec;
+import zipkin.Span;
+import zipkin.storage.AsyncSpanConsumer;
+import zipkin.storage.Callback;
+
+/**
+ * Reports spans to Zipkin, using its {@code POST /spans} endpoint.
+ */
+@AutoValue
+public abstract class URLConnectionReporter implements AsyncSpanConsumer {
+
+  /**
+   * Returns the POST URL for zipkin's <a href="http://zipkin.io/zipkin-api/#/">v1 api</a>, usually
+   * "http://zipkinhost:9411/api/v1/spans".
+   *
+   * <p>This is exposed for those who wish to set this dynamically.
+   */
+  // @FunctionalInterface except library compatible w/ JRE 7
+  public interface PostURL {
```

Comment:
@nicmunroe here's the dynamic thing. This accepts string so that the caller doesn't need to deal with URL's nastiness